src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,244 @@
+import scala.util.{Success, Failure}
+
+
+object WorkflowSubmissionActor {
+  def props(dataSource: SlickDataSource,
+            methodRepoDAO: MethodRepoDAO,
+            googleServicesDAO: GoogleServicesDAO,
+            executionServiceDAO: ExecutionServiceDAO,
+            batchSize: Int,
+            credential: Credential,
+            pollInterval: FiniteDuration): Props = {
+    Props(new WorkflowSubmissionActor(dataSource, methodRepoDAO, googleServicesDAO, executionServiceDAO, batchSize, credential, pollInterval))
+  }
+
+  sealed trait WorkflowSubmissionMessage
+  case object ScheduleNextWorkflowQuery extends WorkflowSubmissionMessage
+  case object LookForWorkflows extends WorkflowSubmissionMessage
+  case class SubmitWorkflowBatch(workflowIds: Seq[Long]) extends WorkflowSubmissionMessage
+
+}
+
+class WorkflowSubmissionActor(val dataSource: SlickDataSource,
+                              val methodRepoDAO: MethodRepoDAO,
+                              val googleServicesDAO: GoogleServicesDAO,
+                              val executionServiceDAO: ExecutionServiceDAO,
+                              val batchSize: Int,
+                              val credential: Credential,
+                              val pollInterval: FiniteDuration) extends Actor with WorkflowSubmission with LazyLogging {
+
+  import context._
+
+  self ! ScheduleNextWorkflowQuery
+
+  override def receive = {
+    case ScheduleNextWorkflowQuery =>
+      scheduleNextWorkflowQuery
+
+    case LookForWorkflows =>
+      getUnlaunchedWorkflowBatch() pipeTo self
+
+    case SubmitWorkflowBatch(workflowIds) =>
+      submitWorkflowBatch(workflowIds) pipeTo self
+
+    case Status.Failure(t) =>
+      logger.error(t.getMessage)
+      scheduleNextWorkflowQuery
+  }
+
+  def scheduleNextWorkflowQuery: Cancellable = {
+    system.scheduler.scheduleOnce(pollInterval, self, LookForWorkflows)
+  }
+}
+
+trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths {
+  val dataSource: SlickDataSource
+  val methodRepoDAO: MethodRepoDAO
+  val googleServicesDAO: GoogleServicesDAO
+  val executionServiceDAO: ExecutionServiceDAO
+  val batchSize: Int
+  val credential: Credential
+  val pollInterval: FiniteDuration
+
+  import dataSource.dataAccess.driver.api._
+
+  //Get a blob of unlaunched workflows, flip their status, and queue them for submission.
+  def getUnlaunchedWorkflowBatch()(implicit executionContext: ExecutionContext): Future[WorkflowSubmissionMessage] = {
+    val unlaunchedWfOptF = dataSource.inTransaction { dataAccess =>
+      //grab a bunch of unsubmitted workflows
+      dataAccess.workflowQuery.findUnsubmittedWorkflows().take(batchSize).result flatMap { wfRecs =>
+        if (wfRecs.isEmpty) {
+          DBIO.successful(wfRecs)
+        } else {
+          //they should also all have the same submission ID
+          val wfsWithASingleSubmission = wfRecs.filter(_.submissionId == wfRecs.head.submissionId)
```

Comment:
So a batch could be less than batchSize, as small as 1 in the worst case.  This is OK, but something to remember.